### PR TITLE
Add PCB functionality back

### PR DESCRIPTION
**Pull request by ff-zhang**
Creates the PCB when necessary and ownership of them is handed to a global `PROCESS_TABLE`.

### CODE REVIEW

**Comment by ff-zhang** [comment]
Let me know what you think is missing from the implementation

File: kernel/src/threading/thread_control_block.rs [hunk 1]
```diff
@@ -69,6 +61,23 @@ pub struct ProcessControlBlock {
...
+    pub fn create() -> Pid {
+        let pid = allocate_pid();
+        let pcb = Self {
+            pid,
+            child_tids: Vec::new(),
+            wait_list: Vec::new(),
+            exit_code: None,
+        };
+        unsafe {
+            PROCESS_TABLE.as_mut().expect("").add(Box::new(pcb));
```
Comment (1whatleytay) [inline kernel/src/threading/thread_control_block.rs:74]: `.unwrap()`?

**Comment by ff-zhang** [comment]
I believe the rest of the places where we would create a PCB aren't merged into `main` yet.

**Review by mckeenicholas — APPROVED** [review]

**Comment by mckeenicholas** [comment]
Processess will have to be removed as well when the thread is reaped.

**Pull request merged**